sdk/webpubsub/azure-webpubsub-upstream/azure/messaging/webpubsub/upstream/exceptions.py

```diff
@@ -0,0 +1,87 @@
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Created on Sun Sep 26 2021
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root
+# for license information.
+# --------------------------------------------------------------------------
+
+__all__ = [
+    "BaseError",
+    "ContextBasedError",
+    "InvalidEventTypeError",
+    "InvalidEventNameError",
+    "InvalidRequestError",
+    "HandlerNotFoundError",
+]
+
+from http import HTTPStatus
+
+
+class BaseError(Exception):
```

Comment:
This exception hierarchy needs to be re-evaluated. Most things should probably be `ValueError`s.